weave/trace/weave_client.py

```diff
@@ -1568,12 +1568,16 @@ def _add_runnable_feedback(
         if not isinstance(runnable_ref, (OpRef, ObjectRef)):
             raise TypeError(f"Invalid scorer op ref: {runnable_ref_uri}")
 
+        from weave.scorers.utils import WeaveScorerResult
+
         # Prepare the result payload - we purposely do not map to refs here
         # because we prefer to have the raw data.
-        results_json = to_json(output, self._project_id(), self)
-
-        # # Prepare the supervision payload
+        if isinstance(output, WeaveScorerResult):
```

Comment:
Please add a note that this is a super hack and should be moved somehow into to_json or something else. We should not need to look for `WeaveScorerResult` as customer defined scores could return pydantics

Reply:
I think that adds to the hackiness and risk. The "correct" solution would be to fix to_json (and more generally our serialization layer) to be clear, centralized, and well defined. I am hesitant increase the surface of this particular shim

Reply:
Let's do this:
https://github.com/wandb/weave/commit/e16aaea3e5ef6288d90e285e78a7e680ecce9ab0

Reply:
@tssweeney wdyt ^?